giskard_vision/core/detectors/metadata_scan_detector.py

```diff
@@ -0,0 +1,199 @@
+from typing import Any, Callable, List, Sequence
+
+import numpy as np
+import pandas as pd
+
+from giskard_vision.core.detectors.base import (
+    DetectorVisionBase,
+    IssueGroup,
+    ScanResult,
+)
+from giskard_vision.core.tests.base import MetricBase
+from giskard_vision.utils.errors import GiskardImportError
+
+
+class MetaDataScanDetector(DetectorVisionBase):
+    """
+    Detector based on Giskard scan that looks for issues based on metadata
+
+    Args:
+        surrogate_function: function
+            Function to transform the output of the model and the ground truth into one value
+            that will be used by the scan
+        metric: function
+            Metric to evaluate the prediction with respect to the ground truth
+        type_task: str
+            Type of the task for the scan, ["regression", "classification"]
+    """
+
+    type_task: str = "classification"
+    surrogate_function: Callable = None
+    metric: MetricBase = None
+    metric_type: str = "relative" if type_task == "regression" else "absolute"
```

Comment:
This won't work if you wanted to do this:
```py
class Foo:
    foo = True
    bar = "A" if foo else "B"
    
class Bar(Foo):
    foo = False
    
bar = Bar()
bar.bar # -> will give "A" not "B"
```

You'd need to put it in `__init__`